sdk/dotnet/Ad/Application.cs

```diff
@@ -0,0 +1,222 @@
+
+        /// <summary>
+        /// Is this Azure AD Application available to other tenants? Defaults to `false`.
+        /// </summary>
+        [Output("availableToOtherTenants")]
+        public Output<bool?> AvailableToOtherTenants { get; private set; } = null!;
+
+        /// <summary>
+        /// The URL to the application's home page. If no homepage is specified this defaults to `https://{name}`.
+        /// </summary>
+        [Output("homepage")]
+        public Output<string> Homepage { get; private set; } = null!;
+
+        /// <summary>
+        /// A list of user-defined URI(s) that uniquely identify a Web application within it's Azure AD tenant, or within a verified custom domain if the application is multi-tenant.
+        /// </summary>
+        [Output("identifierUris")]
+        public Output<ImmutableArray<string>> IdentifierUris { get; private set; } = null!;
+
+        /// <summary>
+        /// The display name for the application.
+        /// </summary>
+        [Output("name")]
+        public Output<string> Name { get; private set; } = null!;
+
+        /// <summary>
+        /// Does this Azure AD Application allow OAuth2.0 implicit flow tokens? Defaults to `false`.
+        /// </summary>
+        [Output("oauth2AllowImplicitFlow")]
+        public Output<bool?> Oauth2AllowImplicitFlow { get; private set; } = null!;
+
+        /// <summary>
+        /// A list of URLs that user tokens are sent to for sign in, or the redirect URIs that OAuth 2.0 authorization codes and access tokens are sent to.
+        /// </summary>
+        [Output("replyUrls")]
+        public Output<ImmutableArray<string>> ReplyUrls { get; private set; } = null!;
+
+
+        /// <summary>
+        /// Create a Application resource with the given unique name, arguments, and options.
+        /// </summary>
+        ///
+        /// <param name="name">The unique name of the resource</param>
+        /// <param name="args">The arguments used to populate this resource's properties</param>
+        /// <param name="options">A bag of options that control this resource's behavior</param>
+        public Application(string name, ApplicationArgs? args = null, CustomResourceOptions? options = null)
+            : base("azure:ad/application:Application", name, args, MakeResourceOptions(options, ""))
+        {
+        }
+
+        private Application(string name, Input<string> id, ApplicationState? state = null, CustomResourceOptions? options = null)
+            : base("azure:ad/application:Application", name, state, MakeResourceOptions(options, id))
+        {
+        }
+
+        private static CustomResourceOptions MakeResourceOptions(CustomResourceOptions? options, Input<string>? id)
+        {
+            var defaultOptions = new CustomResourceOptions
+            {
+                Id = id,
+                Version = Utilities.Version,
+            };
+            return (CustomResourceOptions)ResourceOptions.Merge(defaultOptions, options);
```

Comment:
not certain if this is intentional, or what the implications are.  However, the merging order here differs from js/ts. in js/ts the version specified in 'options' overrides Utilities.version (which is what you have here).  However, the 'id' supplied overrides the one in 'options' (which does not happen here).

Reply:
For normal resource instantiations, the behavior will be the same as in js/ts. For `Get`, if the user specifies an ID in the options bag, that ID will be used instead of the ID specified in the parameter. Happy to change this if that inconsistency feels wrong.

Reply:
i would def prefer to be as consistent in behavior as possible. :)  tnx!